rom/dev/src/flow/cold_reset/fw_processor.rs

```diff
@@ -735,4 +809,78 @@ impl FirmwareProcessor {
 
         Ok(())
     }
+
+    /// Retrieve the fw image from the recovery interface and store it in the mailbox sram.
+    ///
+    /// # Arguments
+    /// * `dma` - DMA driver
+    /// * `soc_ifc` - SOC Interface
+    ///
+    /// # Returns
+    /// * `()` - Ok
+    ///   Error code on failure.
+    fn retrieve_image_from_recovery_interface(
+        dma: &mut Dma,
+        soc_ifc: &mut SocIfc,
+    ) -> CaliptraResult<u32> {
+        let rri_base_addr: u64 = soc_ifc.recovery_interface_base_addr();
+        const PROT_CAP_2_OFFSET: u32 = 0xC;
+        const DEVICE_STATUS_0: u32 = 0x30;
```

Comment:
If this is only accessible via DMA, I'm tempted to make a DMA generator, along the same lines as the RDL / Registers autogenerator. But for now, this is okay.